core/src/main/scala/shapeless/syntax/records.scala

```diff
@@ -81,9 +80,17 @@ final class RecordOps[L <: HList](val l : L) extends AnyVal with Serializable {
    */
   def -[V, Out <: HList](k: Witness)(implicit remover : Remover.Aux[L, k.T, (V, Out)]): Out = remover(l)._2
 
+  /**
+   * Computes the union between this record and another record.
+   */
   def merge[M <: HList](m: M)(implicit merger: Merger[L, M]): merger.Out = merger(l, m)
```

Comment:
Both union and intersection are combining records, it might be better be rename this `def merge` to a `def union`, and `shapeless.ops.record.Merger` to `shapeless.ops.record.Unifier`?